server/internal/infrastructure/memory/project.go

```diff
@@ -0,0 +1,137 @@
+package memory
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/reearth/reearth-cms/server/internal/usecase"
+	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/project"
+	"github.com/reearth/reearth-cms/server/pkg/rerror"
+)
+
+type Project struct {
+	lock sync.Mutex
+	data map[id.ProjectID]*project.Project
+	f    repo.WorkspaceFilter
+}
+
+func NewProject() repo.Project {
+	return &Project{
+		data: map[id.ProjectID]*project.Project{},
+	}
+}
+
+func (r *Project) Filtered(f repo.WorkspaceFilter) repo.Project {
+	return &Project{
+		// note data is shared between the source repo and mutex cannot work well
+		data: r.data,
+		f:    r.f.Merge(f),
+	}
+}
+
+func (r *Project) FindByWorkspace(_ context.Context, id id.WorkspaceID, _ *usecase.Pagination) ([]*project.Project, *usecase.PageInfo, error) {
+	// TODO: implement pagination
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	var result []*project.Project
+	for _, d := range r.data {
+		if d.Workspace() == id {
+			result = append(result, d)
+		}
+	}
```

Comment:
```suggestion
	result := r.data.FindAll(func(k id.ProjectID, v *project.Project) bool {
		return v.Workspace() == id
	})
```

Refer to https://github.com/reearth/reearth-cms/blob/main/server/pkg/util/map.go#L66